### ISSUE TITLE
Add a button for "Not Accept" and add a control in grav pages with cookes settings

### ISSUE THREAD

**Issue by XaRz** [naucon/grav-plugin-cookieconsent#6]
As I've learned with the new GDPR needs two new things relative to cookie consent:

1. User must be give the opt-in option for consent the cookie message. This means that need to be a button saying "I don't accept".
2. This setting must be available in the web for revoquing any time if user decides so. This means that when user consent the cookie setting, perhaps cold be great to be a permanent web control (radiobox or an ON-OFF control) for revoque the cookie consent.

Perhaps something like this? https://www.cookielaw.org/blog/2016/5/13/the-gdpr,-cookie-consent-and-customer-centric-privacy/
(Follow the link cookie settings)
regards.

**Comment by naucon** [comment]
Hi @XaRz thats a difficult topic.

the cookie consent lib has already a option to dismiss or opt-out cookies.
To display the button you could add `"type": "opt-out"` to the js initialise options.

Problem is that there is no functionality behind it. It had to be implemented in callback hook described here:
https://cookieconsent.insites.com/documentation/disabling-cookies/

Neither the cookie consent lib nor this plugin can implemented that logic for you - because it don't know which cookie-based services you might use.

There are open source projects that addresses this problem - like this one for advertiser https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework

If you have a solution to this problem I'am open for suggestion.

**Comment by sunyatasattva** [comment]
Well, you could have a customizable code-editor type field where you could add your own callback, for instance…

**Comment by f3bruary** [comment]
You don't need an Opt-Out button. To comply with GDPR I suggest this:

The cookie notice must have an ACCEPT and REFUSE button.  After the button is clicked, a cookie must be set like "cookie_consent_given" with the value 1/0, or true/false, depending on which button is clicked.

Then you should add a html block in the backend. Here the admin can set the analytics/pixel/tracking/etc scripts.

When consent is given, the scripts are loaded (either through ajax, or upon refresh).

I personally use this method with another plugin on a WordPress site. The script is simply a datalayer that gets pushed to Google Tag Manager. Based on the value of the cookie, I can load certain tags or not.

Don't forget, you only need consent for tracking cookies. Not functional cookies. If you set anonymize_ip to true, and disabled user ID's, then analytics won't track visitors. It will just aggregrate data and you don't need consent for that.

You need consent for conversion tracking, remarketing, facebook pixel, etc. Cookies that track specific user actions on your site.

So in short:

1. Add refuse button
2. Add html block for scripts
3. Insert scripts in front-end page if consent is given

Users can write in their Privacy Policy how visitors can opt-out. This can be through email or phone or whatever. Admin can then delete whatever data they have saved from that specific user.

Policy should contain links to Google/Facebook documentation on how to opt-out from their respective services. 

For example:
https://www.hotjar.com/legal/compliance/opt-out
https://chrome.google.com/webstore/detail/google-analytics-opt-out/fllaojicojecljbmefodhfapmkghcbnh?hl=en
https://www.facebook.com/help/568137493302217

**Comment by rluetke** [comment]
I've added accept/deny/revoke buttons, according to the functions of the underlying JavaScript lib, see pull request #11.

Regards, Ralf
